### DISCUSSION TITLE
Midi

### DISCUSSION THREAD

**Discussion by extremesecrecy** [Builty/TonexOneController#46] [Ideas]
I saw the diagram for the Featherwing, but, how do you connect it with a TRX cable?

**Comment by Builty** [comment]
Hi,
Midi requires both a connector (Din or TRS) and also a small circuit with a device called an optocoupler.
This device isolates the devices for safety.

If you want to use a TRS cable, you could use the Featurewing board, but instead of soldering in the Din sockets, solder in TRS socket instead. This is covered in the Featurewing doc here:
https://github.com/Builty/TonexOneController/blob/main/datasheets/adafruit-midi-featherwing.pdf

If you don't want to use the Featurewing at all, then it would still require the optocoupler circuit to be somewhere. The above doc also shows the optocoupler circuit.
